### DISCUSSION TITLE
Rust analyzer doesnt work in helix

### DISCUSSION THREAD

**Discussion by kittech0** [helix-editor/helix#6663] [Q&A] [answered]
When trying to edit rust files, helix doesnt show any errors from code, nor auto completes it. The rust analyzer works and is in path
![image](https://user-images.githubusercontent.com/73485099/230767093-8fa07d42-c4d1-46dc-8630-bb7a6726c212.png)
![image](https://user-images.githubusercontent.com/73485099/230767209-3fd879a7-8592-4902-b283-64d1e080def0.png)

**Comment by kirawi** [comment]
What does it say when you open your log via `:log-open`?

**Reply by kittech0** [reply]
[helix.log](https://github.com/helix-editor/helix/files/11186087/helix.log)

**Reply by the-mikedavis** [reply, marked as answer]
The line in the log

```
2023-04-09T20:37:50.009 helix_lsp::transport [ERROR] err <- "error: toolchain 'stable-x86_64-unknown-linux-gnu' is not installed\n"
```

is printed to stderr by the language server. This should be the same output as `rustup run stable rust-analyzer` if you ran that by itself in a terminal. You'll need to install the stable toolchain

**Reply by kittech0** [reply]
> The line in the log
> 
> ```
> 2023-04-09T20:37:50.009 helix_lsp::transport [ERROR] err <- "error: toolchain 'stable-x86_64-unknown-linux-gnu' is not installed\n"
> ```
> 
> is printed to stderr by the language server. This should be the same output as `rustup run stable rust-analyzer` if you ran that by itself in a terminal. You'll need to install the stable toolchain

thanks, i forgot i had rust nightly installed
